src/main/webapp/creatives/maps/simid_map_creative.js

```diff
@@ -154,72 +181,162 @@ export default class SimidMapCreative extends BaseSimidCreative {
- * @param {!Object} results An array of Place Results from the search query.
- * @param {!google.maps.places.PlacesServiceStatus} status The status returned 
- *  by the PlacesService on the completion of its searches.
- * @private 
- */
-displayResults_(results, status) {
+  /**
+   * Displays the closest advertisement's locations to a user's current location.
+   * @param {!Object} results An array of Place Results from the search query.
+   * @param {!google.maps.places.PlacesServiceStatus} status The status returned 
+   *  by the PlacesService on the completion of its searches.
+   * @private 
+   */
+  displayResults_(results, status) {
     if (status == google.maps.places.PlacesServiceStatus.OK) {
       for (let i = 0; i < DEFAULT_LOCATION_NUM_DISPLAYED; i++) {
         this.placeMapMarker_(results[i]);
       }
+      this.activeLocation_ = results[0].geometry.location;
+      this.displayDirections_(this.activeLocation_, this.currentLocation_);
+    } else {
+      const statusErrorMessage = {
+        message: "ERROR: Places Service Status was: "+status,
+      };
+      this.simidProtocol.sendMessage(CreativeMessage.LOG, statusErrorMessage);
+      this.playContent_();
     }
-}
+  }
 
-/**
- * Creates and displays a marker on the map representing a given place.
- * @param {!Object} place A Place Result object.
- * @private 
- */
-placeMapMarker_(place) {
-  const placeIcon = {
-    url: this.markerImage_,
-    scaledSize: new google.maps.Size(MARKER_SIZE, MARKER_SIZE)
-  };
-  const placeMarker = new google.maps.Marker({
-    map: this.map_,
-    position: place.geometry.location,
-    icon: placeIcon
-  });
-}
+  /**
+   * Creates and displays a marker on the map representing a given place.
+   * @param {!Object} place A Place Result object.
+   * @private 
+   */
+  placeMapMarker_(place) {
+    const placeIcon = {
+      url: this.markerImage_,
+      scaledSize: new google.maps.Size(MARKER_SIZE, MARKER_SIZE)
+    };
+    const placeMarker = new google.maps.Marker({
+      map: this.map_,
+      position: place.geometry.location,
+      icon: placeIcon
+    });
+    ///Recalculate directions if new marker is clicked.
+    placeMarker.addListener('click', () => {
+      this.activeLocation_ = place.geometry.location;
+      this.displayDirections_(this.currentLocation_, this.activeLocation_);
+    });
+  }
+
+  /**
+   * Displays the route between the starting loaction and destination
+   * based off of the selected travel mode.
+   * @param {!google.maps.LatLng} destination The LatLng coordinates of the destination.
+   * @param {!google.maps.LatLng} startingLocation The LatLng coordinates of the start location.
+   * @private 
+   */
+  displayDirections_(destination, startingLocation) {
+    this.directionsRenderer_.setMap(this.map_);
+    this.calculateRoute_(startingLocation, destination);
+    //If travel method changes, recalculate directions.
+    document.getElementById("travel_method").addEventListener("change", () => {
```

Comment:
Does this add a new event listener every time you display directions?
So if you click back and forth between different locations a few times, then pick a new travel method, will calculateRoute_ run a bunch of times in a row?

Reply:
Yes it would get called multiple times-- the goal is to dynamically update the map so that whenever the user makes a change the map reflects that, I wasn't sure if there was another straightforward way to accomplish this

Reply:
Can you add the listener just once when you create the button the first time, then use class variables instead of passing them through?

Reply:
Great idea-- that's what was causing the bug I found this morning!